vnext/template/cs-app/proj/MyApp.csproj

```diff
@@ -144,6 +144,22 @@
   <ImportGroup Label="ReactNativeWindowsPropertySheets">
     <Import Project="$(ReactNativeWindowsDir)\PropertySheets\External\Microsoft.ReactNative.Uwp.CSharpApp.props" Condition="Exists('$(ReactNativeWindowsDir)\PropertySheets\External\Microsoft.ReactNative.Uwp.CSharpApp.props')" />
   </ImportGroup>
+  <Choose>
+    <When Condition="'$(UseWinUI3)'!='true'">
+      <ItemGroup>
+        <PackageReference Include="Microsoft.UI.Xaml">
+          <Version>$(WinUI2xVersion)</Version>
+        </PackageReference>
+      </ItemGroup>
+    </When>
+    <Otherwise>
+      <ItemGroup>
+        <PackageReference Include="Microsoft.WindowsAppSDK">
+          <Version>$(WinUI3Version)</Version>
+        </PackageReference>
+      </ItemGroup>
+    </Otherwise>
+  </Choose>
```

Comment:
I forget where we landed but judging by the code in the CLI that you are removing that used to synchronize winui dialect, this code should go away and be replaced with a static version fed from the init CLI.
in other words, it sounds like with this PR we are removing the ability to switch between winui2 and winui3 after you create the project (i.e. at build time), that is: once you create a project, you can no longer switch. If that's the case, this code can be simplified to just stamp out the right package/version based on whether they passed --useWinUI3 or not

Reply:
@asklar We're also looking to remove winui3 support from UWP altogether https://github.com/microsoft/react-native-windows/issues/8133, I'm guessing now is too early? Since we're building a WinAppSDK app when --useWinUI3 is specified,  how would this work if we do decide to keep it?

Reply:
Synced offline, will be removing winui3 support from templates in this PR.